### PR TITLE
Add `nvim-tree` plugin integration

### PR DESCRIPTION
**Pull request by jordykoppen**
Hey, I'm back with another addition :)

| Before  | After |
| ------------- | ------------- |
| image | image |

It's more of a quick implementation to get rid of the white border in between. So some suggestions on where to apply what colors would be appreciated!

### CODE REVIEW

**Comment by antoineco** [comment]
I'm interested in this as well.

If I may comment on the changes, all this re-indentation probably belongs in a separate PR. Right now it's really difficult to see what has changed and what has just been re-indented.

**Comment by jordykoppen** [comment]
Ah, the re-indentation was not intentional. Let me fix it 😄

**Comment by sam4llis** [comment]
Hi, thanks for adding to the plugin again!

I agree with @antoineco with respect to the indentation changes. For what its worth, I format using the `sumneko_lua` LSP (with two spaces per indentation level). I think I actually prefer to highlight groups in the first picture (excluding the white border, obviously) as on first glance it looks like file directory highlights already link to the built-in `Directory` highlight group.

If you're interested in adding extra functionality to this integration as well, I believe I've seen some colorschemes have a darker background for the `nvim-tree` window. I'm not sure if this is a built-in functionality inside `nvim-tree`, but it might be worth looking into :) If you don't want to look into this, open up an issue and I'll have a look into it when I have the time!

Once again, thanks for your contributions to the Tundra theme. They're really appreciated, especially when its additions for plugins I don't use!

**Comment by antoineco** [comment]
Here are the highlight groups that control the background of nvim-tree windows: https://github.com/sainnhe/gruvbox-material/issues/139#issuecomment-1239121715

**Comment by jordykoppen** [comment]
Thanks for the suggestions guys!

I've fixed the indentation and removed the highlight groups to keep using the `Directory` highlight group. I did however update the background to be darker, and improve border colors:
image

File: lua/nvim-tundra/stylesheet/arctic.lua [hunk 1]
```diff
@@ -1,5 +1,5 @@
-package.loaded['nvim-tundra.palette.arctic'] = nil
-local cp = require('nvim-tundra.palette.arctic')
+package.loaded["nvim-tundra.palette.arctic"] = nil
+local cp = require("nvim-tundra.palette.arctic")
```
Comment (antoineco) [inline lua/nvim-tundra/stylesheet/arctic.lua]: Small suggestion to revert unrelated changes:

```suggestion
package.loaded['nvim-tundra.palette.arctic'] = nil
local cp = require('nvim-tundra.palette.arctic')
```

File: lua/nvim-tundra/stylesheet/arctic.lua [hunk 2]
```diff
@@ -41,9 +41,9 @@ stylesheet.syntax = {
 
 stylesheet.syntax.builtin = {
   const = cp.orange._500, -- Built-in constants, e.g., `nil` in Lua.
-  func  = cp.orange._500, -- Built-in functions, e.g., `print` in Lua.
-  type  = cp.red._500, -- Built-in types, e.g., `i32` in Rust.
-  var   = cp.orange._500, -- Language-defined variables, e.g., `self` in Python.
+  func = cp.orange._500, -- Built-in functions, e.g., `print` in Lua.
+  type = cp.red._500, -- Built-in types, e.g., `i32` in Rust.
+  var = cp.orange._500, -- Language-defined variables, e.g., `self` in Python.
```
Comment (antoineco) [inline lua/nvim-tundra/stylesheet/arctic.lua]: Small suggestion to revert unrelated changes:

```suggestion
  func  = cp.orange._500, -- Built-in functions, e.g., `print` in Lua.
  type  = cp.red._500, -- Built-in types, e.g., `i32` in Rust.
  var   = cp.orange._500, -- Language-defined variables, e.g., `self` in Python.
```

File: lua/nvim-tundra/stylesheet/arctic.lua [hunk 3]
```diff
@@ -64,6 +64,27 @@ stylesheet.diff = {
...
+  filename = {
+    normal = stylesheet.fg.normal,
+    open = cp.sky._500,
+  },
+  ignored = cp.gray._500,
+  caret = cp.orange._500,
+  statusline = { fg = stylesheet.fg.statusline, bg = stylesheet.bg.transparent },
+  symlink = cp.red._400,
+  border = { bg = stylesheet.bg.normal, fg = stylesheet.bg.normal },
+}
```
Comment (antoineco) [inline lua/nvim-tundra/stylesheet/arctic.lua]: It seems like only `filetree.filename.normal`, `filetree.bg` and `filetree.border` are actually used in this changeset. Is that intentional?

Comment (jordykoppen) [reply lua/nvim-tundra/stylesheet/arctic.lua]: @antoineco sharp! I've removed the unused properties, without altering the styling this time 😂

File: lua/nvim-tundra/base_configuration.lua [hunk 4]
```diff
@@ -32,6 +32,7 @@ local base_configuration = {
   plugins = {
     lsp = true,
     treesitter = true,
+    nvimtree = true,
```
Comment (sam4llis) [inline lua/nvim-tundra/base_configuration.lua]: This should be false by default. I think only 'base' plugins such as native LSP and Treesitter should be enabled by default.

```suggestion
    nvimtree = false,
```

File: lua/nvim-tundra/hl_group/plugins/nvimtree.lua [hunk 5]
```diff
@@ -0,0 +1,10 @@
+local M = {}
+
+M.map = function(ss, opts)
+  return {
+    NvimTreeNormal = { fg = ss.filetree.fg, bg = ss.filetree.bg },
+    NvimTreeVertSplit = ss.filetree.border,
```
Comment (sam4llis) [inline lua/nvim-tundra/hl_group/plugins/nvimtree.lua]: Change to use base stylesheet groups instead of a new `filetree` group. I've also changed the background colour of the `nvim-tree` window so that there is a less drastic contrast between it and normal windows.

Before: 
image

After:
image


```suggestion
    NvimTreeNormal = { fg = ss.fg.normal, bg = ss.bg.dimmed },
    NvimTreeVertSplit = { fg = ss.bg.dimmed, bg = ss.bg.dimmed },
```

File: lua/nvim-tundra/stylesheet/arctic.lua [hunk 6]
```diff
@@ -64,6 +64,12 @@ stylesheet.diff = {
   text = cp.indigo._800,
 }
 
+stylesheet.filetree = {
+  bg = stylesheet.bg.floating,
+  fg = stylesheet.fg.normal,
+  border = { bg = stylesheet.bg.normal, fg = stylesheet.bg.normal },
+}
+
```
Comment (sam4llis) [inline lua/nvim-tundra/stylesheet/arctic.lua]: I think this should be removed from the stylesheet. I want to keep this stylesheet as simple as possible. Check out the suggestion in `nvim-tree.lua` for my proposal!

```suggestion
```

**Review by sam4llis — CHANGES_REQUESTED** [review]
Looking good! Just a couple of changes that I think should be addressed. Also, it might be worth adding some information about `nvim-tree` support into the `README` of this colorscheme.

**Comment by jordykoppen** [comment]
@sam4llis I've committed your suggestions and updated the README 👍

**Comment by sam4llis** [comment]
Great, I'm happy with this integration now if you are as well? I'll likely add it to the `main` branch for version `0.1` of Tundra, alongside some other niceties :)

**Comment by jordykoppen** [comment]
Let's go! 😄
